src/lib/Value/Storage/ItemList.php

```diff
@@ -80,6 +80,15 @@ public function slice(int $offset, ?int $length = null): ItemListInterface
         return new self(array_slice($this->items, $offset, $length));
     }
 
+    public function first(): ItemInterface
+    {
+        if (empty($this->items)) {
+            throw new ItemNotFoundException();
```

Comment:
Should be `OutOfBoundsException` instead.

As discussed privately, I'm generally against using exceptions for cases like this, as empty collection is a valid state and something to be expected. What's more, we're forcing consumers to either handle the exception (performance penalty) or check if the collection is empty beforehand (possible performance penalty).

Considering the alternative - allow `null` to be returned and force consumers to check against `null` - in my opinion it's superior code-wise, performance-wise and more friendly to static analysis.